add qwen 3.8 blog (#307)
vllm-project/vllm-blog-source: _posts/2026-08-12-qwen3.8.md

---
layout: post
title: "Day 0 Support for Qwen3.8-2.4T-A95B on vLLM"
author: "vLLM Team and Inferact"
summary: "Day-0 vLLM support for Qwen3.8-2.4T-A95B: a 2.4-trillion-parameter hybrid MoE model served out of the box, with FP8/BF16 checkpoints plus NVFP4 and MXFP4 quantized weights, and co-developed kernels on NVIDIA and AMD hardware."
image: /assets/figures/2026-08-12-qwen3.8/social-preview.png
social_image: /assets/figures/2026-08-12-qwen3.8/social-preview.png
tags:
  - model-support
  - quantization
  - moe
  - performance
  - hardware
---

> We are announcing Day-0 vLLM support for Qwen3.8-2.4T-A95B. This is the first model from the Qwen family to bring a Qwen-Max-class model to open-weight release.

Reviewer: ```suggestion
We are announcing Day-0 vLLM support for Qwen3.8-2.4T-A95B. This is the first model from the Qwen family to bring a Qwen-class model to open-weight release.
```

> Qwen 3.8 Max is built on the Qwen 3.5 architecture and runs on vLLM out of the box. In addition to the official FP8 and BF16 checkpoints, Inferact has released MXFP4 and NVFP4-quantized weights that match full-precision quality while significantly reducing memory and bandwidth overhead.

Reviewer 2: ```suggestion
Qwen3.8-2.4T-A95B is built on the Qwen 3.5 architecture and runs on vLLM out of the box. In addition to the official FP8 and BF16 checkpoints, Inferact has released MXFP4 and NVFP4-quantized weights that match full-precision quality while significantly reducing memory and bandwidth overhead.
```

---
layout: post
title: "Day 0 Support for Qwen3.8-2.4T-A95B on vLLM"
author: "vLLM Team and Inferact"

> summary: "Day-0 vLLM support for Qwen 3.8 Max (Qwen3.8-2.4T-A95B): a 2.4-trillion-parameter hybrid MoE model served out of the box, with FP8/BF16 checkpoints plus NVFP4 and MXFP4 quantized weights, and co-developed kernels on NVIDIA and AMD hardware."

Reviewer 2: ```suggestion
summary: "Day-0 vLLM support for Qwen3.8-2.4T-A95B: a 2.4-trillion-parameter hybrid MoE model served out of the box, with FP8/BF16 checkpoints plus NVFP4 and MXFP4 quantized weights, and co-developed kernels on NVIDIA and AMD hardware."
```

tags:
  - model-support
  - quantization
  - moe
  - performance
  - hardware
---

> We are announcing Day-0 vLLM support for Qwen 3.8 Max (Qwen3.8-2.4T-A95B to be specific). This is the first model from the Qwen family to bring a Qwen-Max-class model to open-weight release.

Reviewer 2: ```suggestion
We are announcing Day-0 vLLM support for Qwen3.8-2.4T-A95B. This is the first model from the Qwen family to bring a Qwen-Max-class model to open-weight release.
```

Qwen 3.8 Max is built on the Qwen 3.5 architecture and runs on vLLM out of the box. In addition to the official FP8 and BF16 checkpoints, Inferact has released MXFP4 and NVFP4-quantized weights that match full-precision quality while significantly reducing memory and bandwidth overhead.

> Qwen3.8-Max is a 2.4-trillion-parameter sparse MoE model featuring 512 experts. Within its 92-layer hybrid backbone, full attention is applied at every 4th layer while the remaining 69 layers run linear attention. As one of the largest open-weight models released to date, running inference requires at least two NVIDIA B300 / AMD MI355X nodes (or a single node for the FP4 quantized version).

Reviewer 2: ```suggestion
Qwen3.8-2.4T-A95B is a 2.4-trillion-parameter sparse MoE model featuring 512 experts. Within its 92-layer hybrid backbone, full attention is applied at every 4th layer while the remaining 69 layers run linear attention. As one of the largest open-weight models released to date, running inference requires at least two NVIDIA B300 / AMD MI355X nodes (or a single node for the FP4 quantized version).
```

## TL;DR

> - **Day-0 support:** Qwen 3.8 Max reuses the Qwen 3.5 architecture and runs on vLLM from day one with no architecture changes required.

Reviewer 2: ```suggestion
- **Day-0 support:** Qwen3.8-2.4T-A95B reuses the Qwen 3.5 architecture and runs on vLLM from day one with no architecture changes required.
```

## Deployment tips

The Qwen 3.8 model card recommends the following generation parameters for optimal performance:

```
temperature=1.0, top_p=0.95, top_k=20, min_p=0.0, presence_penalty=0.0, repetition_penalty=1.0
```

Here is a Python client snippet to query the model once the vLLM server is running. Because Qwen 3.8 is a reasoning model, ensure you allocate a sufficient token budget for agentic workflows by setting a high `max_tokens` value.

```python
from openai import OpenAI

client = OpenAI(api_key="EMPTY", base_url="http://localhost:8000/v1", timeout=3600)

resp = client.chat.completions.create(
    model="Qwen/Qwen3.8-2.4T-A95B",
    messages=[{"role": "user", "content": "Give me three primes above 100."}],
    temperature=1.0, top_p=0.95, max_tokens=128_000,
)
print(resp.choices[0].message.content)
```

## Acknowledgements

> We thank the Qwen team for releasing the model weights as well as their ongoing collaboration, and our hardware partners, NVIDIA and AMD, for their joint engineering contributions. We also thank the Inferact team for delivering quantized checkpoints and end-to-end vLLM integration, as well as the broader vLLM community.

Reviewer 3: ```suggestion
We thank the Qwen team for releasing the model weights as well as their ongoing collaboration, and our hardware partners, NVIDIA and AMD, for their joint engineering contributions. We also thank the Inferact team for delivering quantized checkpoints and end-to-end vLLM integration, as well as the broader vLLM community. Thanks for our inference partners, including DigitalOcean, Together AI, who helped with early testing.
```
